### ISSUE TITLE
Xenos devouring through fog

### ISSUE THREAD

**Issue by InterroLouis** [tgstation/TerraGov-Marine-Corps#2827]
## Testmerges:
Server-Side Code Modifications: https://tgstation13.org/parsed-logs/basil/server_side_modifications.dm

Round ID: 949
BYOND Version: 512.1464
Server revision compiled on: 
Master commit: c2862d6f94028f8efc34581d8cb4ae35c19bc88d
The following pull requests are currently test merged:
#2821: 'Fixes a sound runtime' by DominikPanic at commit 76f205e490
#2736: 'Movespeed mod and datum charge/crushing' by Rohesie at commit 48826a261c
#2797: 'Reduces the roundstart notransform lock time' by DominikPanic at commit 83b60755ef
#2782: 'Fulton extraction devices' by Rohesie at commit f989905051
#2795: '[WIP] Pillar of Spring' by grinkgo at commit ff2a5ef8af
#2823: 'Fixes powerloader clamp runtimes' by DominikPanic at commit df72aa5602
#2825: 'Incoming mentorhelps now have a different sound' by DominikPanic at commit ec8dfe9057
#2819: 'New PFC armor sprites and new squad color armor markings' by Metroidlover at commit be5cc9c6fa
#2814: '[W.I.P] Interaction rework' by DominikPanic at commit 5f3a5d6e68

Local commit: 4c5cfbf50171dbb11b3322ba9556335a64ab82a3
Server tools version: /tg/station 13 Server v3.2.6.2
## Reproduction:
Basically just have a xeno stand ON the fog tile and devour a marine, then they can take them through the fog despite the fact that they can't go through fog with a marine inside. The issue is because they are already on the fog tile when they devour, it doesn't count as them moving onto the fog tile with a marine in them-recommend disabling devour function while xenos are on a fog tile.

**Comment by DominikPanic** [comment]
@Rohesie
